File: apps/ios/Domain/Tests/DomainTests/PushStateVectors.swift
import Foundation
@testable import Domain

// Wires PushState.swift into FunctionRegistry.
//
// The whole cross-product of the three inputs, because the one that matters —
// blocked vs off — looks identical on screen and only one of them can be fixed
// by tapping the switch.

func registerPushStateVectors() {
    let domain = "push-state"

    FunctionRegistry.register(domain: domain, fn: "pushState") { input in
        let d = input as! [String: Any]
        return pushState(
            supported: (d["supported"] as! NSNumber).boolValue,
            permission: d["permission"] as! String,
            prefEnabled: (d["prefEnabled"] as! NSNumber).boolValue
        ).rawValue
    }

    FunctionRegistry.register(domain: domain, fn: "shouldRegisterAtLaunch") { input in
        let d = input as! [String: Any]
        return shouldRegisterAtLaunch(
            permission: d["permission"] as! String,
            prefEnabled: (d["prefEnabled"] as! NSNumber).boolValue
        )
    }
}

File: apps/android/domain/src/test/kotlin/com/sanvya/app/domain/notifications/PushStateVectors.kt
package com.sanvya.app.domain.notifications

import com.sanvya.app.domain.vectors.FunctionRegistry
import kotlinx.serialization.json.JsonPrimitive
import kotlinx.serialization.json.boolean
import kotlinx.serialization.json.jsonObject
import kotlinx.serialization.json.jsonPrimitive

// Wires PushState.kt into FunctionRegistry.
//
// The whole cross-product of the three inputs, because the one that matters --
// BLOCKED vs OFF -- looks identical on screen and only one of them can be fixed
// by tapping the switch.

private const val DOMAIN = "push-state"

fun registerPushStateVectors() {
    FunctionRegistry.register(DOMAIN, "pushState") { input ->
        val o = input.jsonObject
        JsonPrimitive(
            pushState(
                supported = o.getValue("supported").jsonPrimitive.boolean,
                permission = o.getValue("permission").jsonPrimitive.content,
                prefEnabled = o.getValue("prefEnabled").jsonPrimitive.boolean,
            ).name.lowercase(),
        )
    }

    FunctionRegistry.register(DOMAIN, "shouldRegisterAtLaunch") { input ->
        val o = input.jsonObject
        JsonPrimitive(
            shouldRegisterAtLaunch(
                permission = o.getValue("permission").jsonPrimitive.content,
                prefEnabled = o.getValue("prefEnabled").jsonPrimitive.boolean,
            ),
        )
    }
}
